### PR TITLE
Allow download operations to accept an open file handle

### PR DESCRIPTION
**Pull request by MisterToolbox**
I've run into a case where I'd like to preallocate disk space before S3 downloads begin. Since the download operation opens files in write mode, preallocations are wiped out when the actual download begins. 

If the operation can optionally take the pid of an open file handle, preallocations can be preserved. Otherwise, sending in a file name as a binary works as usual.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -19,3 +19,4 @@ erl_crash.dump
 # Also ignore archive artifacts (built via "mix archive.build").
 *.ez
 /tmp
+mix.lock
```
Comment (benwilson512) [inline .gitignore]: Please don't gitignore this, it makes it incredibly hard to have reproducibility between developers on the project.

File: README.md [hunk 2]
```diff
@@ -1,7 +1,10 @@
 # ExAws.S3
-
 Service module for https://github.com/ex-aws/ex_aws
 
+## IMPORTANT NOTE
+This is a slightly modified fork of [https://github.com/ex-aws/ex_aws_s3].
+Once [the relevant Pull Request](https://github.com/ex-aws/ex_aws_s3/pull/16) is merged in, this will be deleted.
```
Comment (benwilson512) [inline README.md]: Please remove this so I can merge.

File: lib/ex_aws/s3.ex [hunk 3]
```diff
@@ -487,6 +487,18 @@ defmodule ExAws.S3 do
       bucket: bucket,
       path: path,
       dest: dest,
+      object: nil,
+      opts: opts
+    }
+  end
+
+  @spec download_file(bucket :: binary, path :: binary, dest :: binary, object :: map, opts :: download_file_opts) :: __MODULE__.Download.t
+  def download_file(bucket, path, dest, object, opts) do
```
Comment (benwilson512) [inline lib/ex_aws/s3.ex]: What is `object` here and why is it required?

Comment (benwilson512) [reply lib/ex_aws/s3.ex]: I would definitely like to avoid breaking changes to the public API.

File: .gitignore (hunk 1 above)
Comment (jalcine) [reply .gitignore]: Yeah, this is a really important file to have committed.

**Pull request closed without merging**